### DISCUSSION TITLE
what' the path of jars needed by metals?

### DISCUSSION THREAD

**Discussion by passion123** [scalameta/metals#5309] [Q&A] [answered]
my versions as belows:
theia :1.36.0
mvn:3.8.8
scala:2.11.8
metals(scala):1.23.0--1.11.0

i want to debug and run the scala language by theia ,but it always shows the error 
![image](https://github.com/scalameta/metals/assets/9447543/921147b0-dcb9-4773-93da-0fb58c00605b)
but i can use wget to get these jars to be download on the linux ubuntu 20.04 server,is there any one know how to get there jars online or put it to the pathed pointed by offline ? 
btw: i have set the metals version and java_home but it useless,still shows the error message

**Comment by tgodzik** [comment, marked as answer]
Thanks for the question! Any idea why those jars are not being downloaded? Is maven central available within your environment?

One possibility to work around it would be to install coursier locally and that should be picked up by Metals and there wouldn't be any need to download those jars directly. There might still be an issue when downloading the metals server jars if you're behind a proxy or need to use a custom repository to fetch artifacts.

There is a bit of infor about it in https://scalameta.org/metals/docs/troubleshooting/proxy/

**Reply by passion123** [reply]
Thx for your response.I solved it by setting the proxy as your guide file  with **metals.serverProperties**, and the metals' dependencies can be downloaded.But while dubugging the scala file  it shows another error as below:
**Error, couldn't show release notes for Metals 0.11.12
Timeout occured during get request at https://api.github.com/repos/scalameta/metals/releases/tags/v0.11.12
2023.06.06 08:58:27 INFO  running '/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -Dfile.encoding=UTF-8 -Dmaven.multiModuleProjectDirectory=/home/project/myproject -Dmaven.home=/tmp/metals8374571292977612979 -cp /tmp/metals8374571292977612979/maven-wrapper.jar org.apache.maven.wrapper.MavenWrapperMain generate-sources ch.epfl.scala:bloop-maven-plugin:2.0.0:bloopInstall -DdownloadSources=true'
2023.06.06 09:00:35 ERROR Exception in thread "main" java.net.ConnectException: Connection timed out (Connection timed out)
2023.06.06 09:00:35 ERROR 	at java.net.PlainSocketImpl.socketConnect(Native Method)
2023.06.06 09:00:35 ERROR 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
2023.06.06 09:00:35 ERROR 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
2023.06.06 09:00:35 ERROR 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
2023.06.06 09:00:35 ERROR 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
2023.06.06 09:00:35 ERROR 	at java.net.Socket.connect(Socket.java:607)
2023.06.06 09:00:35 ERROR 	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:293)
2023.06.06 09:00:35 ERROR 	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
2023.06.06 09:00:35 ERROR 	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
2023.06.06 09:00:35 ERROR 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
2023.06.06 09:00:35 ERROR 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:203)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1162)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1056)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:189)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1572)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1500)
2023.06.06 09:00:35 ERROR 	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:268)
2023.06.06 09:00:35 ERROR 	at org.apache.maven.wrapper.DefaultDownloader.downloadInternal(DefaultDownloader.java:109)
2023.06.06 09:00:35 ERROR 	at org.apache.maven.wrapper.DefaultDownloader.download(DefaultDownloader.java:97)
2023.06.06 09:00:35 ERROR 	at org.apache.maven.wrapper.Installer.createDist(Installer.java:94)
2023.06.06 09:00:35 ERROR 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:151)
2023.06.06 09:00:35 ERROR 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:76)
2023.06.06 09:00:35 INFO  time: ran 'mvn bloopInstall' in 2m8s
2023.06.06 09:00:35 ERROR Maven command failed: /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -Dfile.encoding=UTF-8 -Dmaven.multiModuleProjectDirectory=/home/project/myproject -Dmaven.home=/tmp/metals8374571292977612979 -cp /tmp/metals8374571292977612979/maven-wrapper.jar org.apache.maven.wrapper.MavenWrapperMain generate-sources ch.epfl.scala:bloop-maven-plugin:2.0.0:bloopInstall -DdownloadSources=true**
my launch file as below:
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  "version": "0.2.0",
  "configurations": [
    {
      "type": "scala",
      "request": "launch",
      "name": "Launch test",
      "mainClass": "com.test.Test",
      "args": [],
      "jvmOptions": []
    },
  ]
}
do u have any ide on solving the debugging scala file ?

**Reply by tgodzik** [reply]
Looks like the error is from importing the build, so it looks like we aren't forwarding the proxy settings to maven command. How do you normally set up maven? As a workaround you can run `mvn ch.epfl.scala:bloop-maven-plugin:2.0.0:bloopInstall -DdownloadSources=true` with the proxy settings
